src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -78,35 +95,85 @@ public async Task<VSReferenceItem[]> HandleRequestAsync(ReferenceParams request,
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var serverKind = LanguageServerKind.CSharp;
-            var referenceParams = new ReferenceParams()
+            // Temporary till IProgress serialization is fixed
+            var requestId = Guid.NewGuid().ToString(); // request.PartialResultToken.Id
+
+            var referenceParams = new CustomReferenceParams()
             {
                 Position = projectionResult.Position,
                 TextDocument = new TextDocumentIdentifier()
                 {
                     Uri = projectionResult.Uri
                 },
                 Context = request.Context,
-                PartialResultToken = request.PartialResultToken
+                PartialResultToken = requestId // request.PartialResultToken
             };
 
-            var result = await _requestInvoker.ReinvokeRequestOnServerAsync<ReferenceParams, VSReferenceItem[]>(
-                Methods.TextDocumentReferencesName,
-                serverKind,
-                referenceParams,
-                cancellationToken).ConfigureAwait(false);
+            var waitForProgressCompletion = new ManualResetEvent(false);
+
+            Task callback(JToken value)
```

Comment:
I'd just create a lambda in the callback request construction to be consistent with how we do this elsewhere